Several versions of llvm can coexist in the archive (at the time of this
writing, 2.8 and 2.9), so pick a version. Let's assume the last version
receives bug fixes quickly, so pick 2.9 for now.

(Ammended 01/30/2012 - Sarvatt)
Switch to 3.0 for ubuntu 12.04
Index: mesa/configure.ac
===================================================================
--- mesa.orig/configure.ac	2012-05-21 18:19:43.639922922 -0400
+++ mesa/configure.ac	2012-05-21 18:19:43.947922933 -0400
@@ -1762,7 +1762,7 @@
     esac
 fi
 if test "x$enable_gallium_llvm" = xyes; then
-    AC_PATH_PROG([LLVM_CONFIG], [llvm-config], [no])
+    AC_PATH_PROG([LLVM_CONFIG], [llvm-config-3.0], [no])
 
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version | sed 's/svn.*//g'`
